modules/index/src/test/java/org/apache/ignite/internal/index/IndexBuilderTest.java

```diff
@@ -162,12 +177,43 @@ private void scheduleBuildIndex(int indexId, int tableId, int partitionId, Colle
         );
     }
 
+    private void scheduleBuildIndexAfterDisasterRecovery(int indexId, int tableId, int partitionId, Collection<RowId> nextRowIdsToBuild) {
+        indexBuilder.scheduleBuildIndexAfterDisasterRecovery(
+                tableId,
+                partitionId,
+                indexId,
+                indexStorage(nextRowIdsToBuild),
+                mock(MvPartitionStorage.class),
+                mock(ClusterNode.class),
+                ANY_ENLISTMENT_CONSISTENCY_TOKEN,
+                ANY_INDEX_CREATION_CATALOG_VERSION
+        );
+    }
+
     private CompletableFuture<Void> listenCompletionIndexBuilding(int indexId, int tableId, int partitionId) {
-        CompletableFuture<Void> future = new CompletableFuture<>();
+        var future = new CompletableFuture<Void>();
 
-        indexBuilder.listen((indexId1, tableId1, partitionId1) -> {
-            if (indexId1 == indexId && tableId1 == tableId && partitionId1 == partitionId) {
-                future.complete(null);
+        indexBuilder.listen(new IndexBuildCompletionListener() {
+            @Override
+            public void onBuildCompletion(int indexId1, int tableId1, int partitionId1) {
```

Comment:
Could we also assert that `onBuildCompletionAfterDisasterRecovery`method is not triggered? The same for `listenCompletionIndexBuildingAfterDisasterRecovery`, but ofc `onBuildCompletion`